src/main/java/jenkins/plugins/logstash/LogstashWriter.java

```diff
@@ -61,8 +62,26 @@
   private final LogstashIndexerDao dao;
   private boolean connectionBroken;
   private final Charset charset;
+  private final EnvVars envVars;
 
   public LogstashWriter(Run<?, ?> run, OutputStream error, TaskListener listener, Charset charset) {
+    this.envVars = null;
+    this.errorStream = error != null ? error : System.err;
+    this.build = run;
+    this.listener = listener;
+    this.charset = charset;
+    this.dao = this.getDaoOrNull();
+    if (this.dao == null) {
+      this.jenkinsUrl = "";
+      this.buildData = null;
+    } else {
+      this.jenkinsUrl = getJenkinsUrl();
+      this.buildData = getBuildData();
+    }
+  }
```

Comment:
This is a duplication of code. Instead just call the new constructor forwarding all arguments and pass null for the envvars

Reply:
OK